proto/teleport/lib/teleterm/v1/service.proto

```diff
@@ -471,6 +473,21 @@ message ListDatabaseUsersResponse {
   repeated string users = 1;
 }
 
+message ListResourcesRequest {
+  string cluster_uri = 1;
+  string start_key = 2;
+  int32 limit = 3;
+  string predicate_expression = 4;
+  // UseSearchAsRoles indicates that the response should include all resources
+  // the caller is able to request access to using search_as_roles
+  bool use_search_as_roles = 5;
+}
```

Comment:
I named it this way because I wanted to communicate that we use [ListResources](https://github.com/gravitational/teleport/blob/16aa45237578f4fb4aa7be253f7902082c13f0fb/lib/auth/auth_with_roles.go#L1766) for these requests.

I use `ListResources` here instead of ListUnifiedResources, because the former supports querying for other kinds of resources (eg: db_server, unified resources does not support this)

I might be thinking too far in the future, but I wanted like a generic naming (ListResourcesRequest instead of `ListDatabaseServersRequest` because I was told we want to add support for kube status in q2, so we will also be using ListResources to fetch kube_servers as well.

I talked with @avatus and confirmed that ListResources api is not going away anytime soon.

Reply:
It's not recommend to reuse requests/responses https://protobuf.dev/best-practices/api#unique-protos.
Instead we could extract just the resource query (not sure about naming) part, something along these lines:
```proto
message ListDatabaseServersRequest {
  string cluster_uri = 1;
  ResourceQueryFilter query = 2;
}

message ResourceQueryFilter {
  string start_key = 1;
  int32 limit = 2;
  string predicate_expression = 3;
  // UseSearchAsRoles indicates that the response should include all resources
  // the caller is able to request access to using search_as_roles
  bool use_search_as_roles = 4;
}
```